lib/kms/wrapper.js

```diff
@@ -0,0 +1,246 @@
+        } else if (sseAlgorithm === 'aws:kms') {
+            const serverSideEncryptionInfo = {
+                cryptoScheme: 1,
+                algorithm: sseAlgorithm,
+                masterKeyId: sseMasterKeyId,
+                mandatory: true,
+            };
+            cb(null, serverSideEncryptionInfo);
+        } else {
+            /*
+             * no encryption
+             */
+            cb(null, null);
+        }
+    }
+
+    static createDataKey(log) {
+        log.debug('creating a new data key');
+        const newKey = common.createDataKey();
+        log.trace('data key created by the kms');
+        return newKey;
+    }
+
+    /*
+     * cb(err, cipheredDataKey: Buffer)
+     */
+    static _cipherDataKey(cryptoScheme,
+                          masterKeyId,
+                          plainTextDataKey,
+                          log, cb) {
+        log.debug('ciphering a data key');
+        client.cipherDataKey(cryptoScheme, masterKeyId, plainTextDataKey, log,
+                             (err, cipheredDataKey) => {
+                                 if (err) {
+                                     log.warn('error from kms',
+                                              { implName, error: err });
+                                     return cb(err);
+                                 }
+                                 log.trace('data key ciphered by the kms');
+                                 return cb(null, cipheredDataKey);
+                             });
+    }
+
+    /*
+     * cb(err, plainTextDataKey: Buffer)
+     */
+    static _decipherDataKey(cryptoScheme, masterKeyId,
+                            cipheredDataKey, log, cb) {
+        log.debug('deciphering a data key');
+        client.decipherDataKey(cryptoScheme, masterKeyId, cipheredDataKey, log,
+                               (err, plainTextDataKey) => {
+                                   if (err) {
+                                       log.warn('error from kms',
+                                                { implName, error: err });
+                                       return cb(err);
+                                   }
+                                   log.trace('data key deciphered by the kms');
+                                   return cb(null, plainTextDataKey);
+                               });
+    }
+
+    /*
+     * cb(err, cipher: ReadWritable.stream)
+     */
+    static _createCipher(cryptoScheme, dataKey, offset, log, cb) {
+        log.debug('creating a cipher');
+        common.createCipher(
+            cryptoScheme, dataKey, offset, log,
+            (err, cipher) => {
+                if (err) {
+                    log.warn('error from kms', { implName, error: err });
+                    return cb(err);
+                }
+                log.trace('cipher created by the kms');
+                return cb(null, cipher);
+            });
+    }
+
+    /*
+     * cb(err, decipher: ReadWritable.stream)
+     */
```

Comment:
maybe call common.createDecipher directly from the code without this extra function layer